internal/blockcreator/blockcreator.go

```diff
@@ -148,13 +149,17 @@ func (b *BlockCreator) Start() {
 				// This may happen when shutting down the server. 'continue' will eventually pick up the stop signal.
 				b.logger.Warnf("block submission to block-replicator failed, dropping block, shutting down, because: %s", err)
 				continue
+
 			case *ierrors.NotLeaderError:
 				b.logger.Warnf("block submission to block-replicator failed, because this node is not a leader: %s", err)
-				//TODO reject or redirect all TXs in the block via the pending-tx component.
-				// If there is another leader then redirect, else reject.
+				// Releasing with an error will reject or redirect all sync TXs in the block via the pending-tx component.
+				// If there is another leader it will redirect, else reject. Async TXs will be removed.
```

Comment:
How does the redirection work? Is it done automatically by the raft or we need to implement it?

Reply:
When a tx promise is waited upon by a request go-routine, and it is release with an error, the handler will process the error; 
- if it is a NotLeader with an alternative leader, it will with redirect (307) the client to the new leader;
- if their is no leader it will reject with a "service unavailable"  (503) 

This is already in place, see:
https://github.com/hyperledger-labs/orion-server/blob/04f19b212c8428d927c214babe390258e540307b/internal/httphandler/tx_handler.go#L30

async requests are dropped at this stage, as we don't forward internally now.